### PR TITLE
Btree indexes are deprecated

### PR DESCRIPTION
**Pull request by Hunterness**

### CODE REVIEW

**Comment by Hunterness** [comment]
Open questions:
- [x] Do we need/want to update in more places?
- [x] add link to migration guide?
- [x] what category? functionality? syntax?
- [x] do we want to specify the btree options?

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 1]
```diff
@@ -75,6 +75,40 @@ Replaced by:
...
+B-tree indexes are deprecated and will need migration before upgrading to 5.0.
+
+a|
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+CREATE CONSTRAINT
+...
+OPTIONS "{" btree-option: btree-value[, ...] "}"
```
Comment (Hunterness) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc:146]: should have this for `CREATE INDEX ...` as well

**Comment by Hunterness** [comment]
[Document deprecation of BTREE indexes](https://trello.com/c/fCrLfaLN/2596-document-deprecation-of-btree-indexes)

**Comment by Lojjs** [comment]
> Open questions:
> 
> * [x]  Do we need/want to update in more places?
> * [x]  add link to migration guide?
> * [x]  what category? functionality? syntax?
> * [x]  do we want to specify the btree options for constraints?

1. Under the assumption that the planner team will not have time to add any index support, I guess this should be enough. At least I don't think we need to update anything else at this point.

2. Since there is no migration chapter for 4.x -> 5.0 yet and we don't know if it will be in the same guide or a separate, I would not add a link now. Maybe we can add a link later when the content about index migration is written.

3. I think **functionality** is most correct, since it is not just about changed syntax

4. Yes, and you did already fix this

So in summary, I think this is ready to go in after a check from the docs team

**Review by Lojjs — APPROVED** [review]

**Comment by Hunterness** [comment]
4. Not what I ment, the question was if we should add the actual values or example of them and not just the generic `btree-option: <btree-value>` thing I added now

what does the docs team think?

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 2]
```diff
@@ -102,6 +102,52 @@ OPTIONS "{" option: value[, ...] "}"
...
+
+a|
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+CREATE BTREE INDEX ...
+----
+a|
+B-tree indexes are deprecated and will need migration before upgrading to 5.0.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
B-tree indexes are deprecated and need to be migrated before upgrading to 5.0.
```

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 3]
```diff
@@ -102,6 +102,52 @@ OPTIONS "{" option: value[, ...] "}"
...
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+CREATE INDEX
+...
+OPTIONS "{" btree-option: btree-value[, ...] "}"
+----
+a|
+B-tree indexes are deprecated and will need migration before upgrading to 5.0.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
B-tree indexes are deprecated and need to be migrated before upgrading to 5.0.
```

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 4]
```diff
@@ -102,6 +102,52 @@ OPTIONS "{" option: value[, ...] "}"
...
+
+a|
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+SHOW BTREE INDEXES
+----
+a|
+B-tree indexes are deprecated and will need migration before upgrading to 5.0.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
B-tree indexes are deprecated and need to be migrated before upgrading to 5.0.
```

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 5]
```diff
@@ -102,6 +102,52 @@ OPTIONS "{" option: value[, ...] "}"
...
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+CREATE CONSTRAINT
+...
+OPTIONS "{" btree-option: btree-value[, ...] "}"
+----
+a|
+Node key and uniqueness constraints with b-tree options are deprecated and will need migration before upgrading to 5.0.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
Node key and uniqueness constraints with b-tree options are deprecated and need to be migrated before upgrading to 5.0.
```

**Review by AlexicaWright — COMMENTED** [review]
Editorial comments only.

**Comment by AlexicaWright** [comment]
As for your question 4, I don't think you need to add the actual values here.

**Review by AlexicaWright — APPROVED** [review]
👍  Thank you @Hunterness !
